vulnfeeds/cmd/cvelist2osv/main.go

```diff
@@ -0,0 +1,290 @@
+	NoSoftware                                 // The CVE had no CPEs relating to software (i.e. Operating Systems or Hardware).
+	NoRepos                                    // The CPE Vendor/Product had no repositories derived for it.
+	NoRanges                                   // No viable commit ranges could be calculated from the repository for the CVE's CPE(s).
+	FixUnresolvable                            // Partial resolution of versions, resulting in a false positive.
+)
+
+var (
+	jsonPath  = flag.String("cve_json", "", "Path to CVEList JSON to examine.")
+	outDir    = flag.String("out_dir", "", "Path to output results.")
+	outFormat = flag.String("out_format", "OSV", "Format to output {OSV}")
+)
+var Logger utility.LoggerWrapper
+var RepoTagsCache git.RepoTagsCache
+var Metrics struct {
+	CNA           string
+	Outcome       ConversionOutcome
+	RefTypesCount map[osvschema.ReferenceType]int
+	Notes         []string
+}
+
+// RefTagDenyList: References with these tags have been found to contain completely unrelated
+// repositories and can be misleading as to the software's true repository,
+// Currently not used for this purpose due to undesired false positives
+// reducing the number of valid records successfully converted.
+var RefTagDenyList = []string{
+	// "Exploit",
+	// "Third Party Advisory",
+	"Broken Link", // Actively ignore these though.
+}
+
+// addIDToNotes prepends a CVE ID to a slice of notes.
+func addIDToNotes(id cves.CVEID, notes []string) []string {
+	var updatedNotes []string
+	for _, note := range notes {
+		updatedNotes = append(updatedNotes, fmt.Sprintf("[%s] %s", id, note))
+	}
+	return updatedNotes
+}
+
+// determineHeuristics examines a CVE and extracts metrics and heuristics about it.
+func determineHeuristics(cve cves.CVE5, v *vulns.Vulnerability) {
+	// CNA based heuristics
+	Metrics.CNA = cve.Metadata.AssignerShortName
+	// TODO(jesslowe): more CNA based analysis
+	// Reference based heuristics
+	// Count number of references of each type
+	refTypeCounts := make(map[osvschema.ReferenceType]int)
+	for _, ref := range v.References {
+		refTypeCounts[ref.Type]++
+	}
+	Metrics.RefTypesCount = refTypeCounts
+	for refType, count := range refTypeCounts {
+		Metrics.Notes = append(Metrics.Notes, fmt.Sprintf("[%s]: Reference Type %s: %d", cve.Metadata.CVEID, refType, count))
+	}
+}
+
+// FromCVE5 creates a Vulnerability from a CVE5 object.
+func FromCVE5(cve cves.CVE5, refs []cves.Reference) (*vulns.Vulnerability, []string) {
+	aliases, related := vulns.ExtractReferencedVulns(cve.Metadata.CVEID, cve.Metadata.CVEID, refs)
+	var err error
+	var notes []string
+	v := vulns.Vulnerability{}
+	v.SchemaVersion = osvschema.SchemaVersion
+	v.ID = string(cve.Metadata.CVEID)
+	v.Summary = string(cve.Containers.CNA.Title)
+	v.Details = cves.EnglishDescription(cve.Containers.CNA.Descriptions)
+	v.Aliases = aliases
+	v.Related = related
+	v.Published, err = cves.ParseCVE5Timestamp(cve.Metadata.DatePublished)
+	if err != nil {
```

Comment:
should we check error first before assigning the value to v.Published/v.Modified? what would the value be when parse failed?